lib/phoenix/test/conn_test.ex

```diff
@@ -464,7 +479,8 @@ defmodule Phoenix.ConnTest do
   """
   @spec recycle(Conn.t) :: Conn.t
   def recycle(conn) do
-    build_conn()
+    build_conn(conn)
```

Comment:
Instead of changing `build_conn`, wouldn't be better to have a function called `copy_host` or `copy_request_uri` or similar? That's how we handle copy_headers, for example.